### PR TITLE
Correctly set single ECS container name

### PR DESCRIPTION
**Pull request by thomas11**
Fixes https://github.com/pulumi/pulumi-awsx/issues/882

Using `container.name`, which is an `Output`, required output-ifying that code path.

### CODE REVIEW

**Review by iwahbe — COMMENTED** [review]
This is a big change to not have any tests. Would it be possible to add some?

I'm not super familiar with aws itself, so I'm hesitant to approve something that I can't check. Others may feel differently.

**Comment by thomas11** [comment]
> This is a big change to not have any tests. Would it be possible to add some?
> 
> I'm not super familiar with aws itself, so I'm hesitant to approve something that I can't check. Others may feel differently.

@iwahbe I did add tests for the functionality I fixed, in `awsx/ecs/container.test.ts`. Granted, coverage for other code paths is still low to none.

**Comment by iwahbe** [comment]
> > This is a big change to not have any tests. Would it be possible to add some?
> > I'm not super familiar with aws itself, so I'm hesitant to approve something that I can't check. Others may feel differently.
> 
> @iwahbe I did add tests for the functionality I fixed, in `awsx/ecs/container.test.ts`. Granted, coverage for other code paths is still low to none.

My bad, I totally missed that. I was looking for files that had test in the name, not the extension.

File: awsx/ecs/containers.ts [hunk 1]
```diff
@@ -18,21 +18,43 @@ import { LogGroupId } from "../cloudwatch/logGroup";
...
-
-  return pulumi.all(result);
+  const computed = containers.apply((c) => {
+    const res: pulumi.Output<schema.TaskDefinitionContainerDefinitionInputs>[] = [];
+    for (const containerName of Object.keys(c)) {
+      const container = c[containerName];
+      res.push(computeContainerDefinition(parent, containerName, container, logGroupId));
+    }
+    return pulumi.all(res);
+  });
```
Comment (mjeffryes) [inline awsx/ecs/containers.ts:56]: nit: could this just be:
```
pulumi.all(Object.entries(c).map([containerName, container] => 
  computeContainerDefinition(parent, containerName, container, logGroupId)
))
```
?

**Review by mjeffryes — APPROVED** [review]
Looks correct to me, all my comments are just style nits and non-blocking

File: awsx/ecs/containers.ts (hunk 1 above)
Comment (thomas11) [reply awsx/ecs/containers.ts]: Nicely functional, thanks!

**Pull request merged**